lisa/nic.py

```diff
@@ -285,37 +325,38 @@ def reload(self) -> None:
         self._initialize()
 
     @retry(tries=15, delay=3, backoff=1.15)
-    def wait_for_sriov_enabled(self) -> None:
+    def check_pci_enabled(self, pci_enabled: bool) -> None:
         lspci = self._node.tools[Lspci]
-
-        # check for VFs on the guest
-        vfs = lspci.get_devices_by_type(constants.DEVICE_TYPE_SRIOV, force_run=True)
-        assert_that(len(vfs)).described_as(
-            "Could not identify any SRIOV NICs on the test node."
-        ).is_not_zero()
-
-        # check if the NIC driver has finished setting up the
-        # failsafe pair, reload if not
-        if not self.get_lower_nics():
-            self.reload()
-        if not self.get_lower_nics():
-            assert_that(self.get_lower_nics()).described_as(
-                "Did not detect any upper/lower sriov paired nics!: "
-                f"upper: {self.get_upper_nics()} "
-                f"lower: {self.get_lower_nics()} "
-                f"unpaired: {self.get_unpaired_devices()}"
-                f"vfs: {','.join([str(pci) for pci in vfs])}"
-            ).is_not_empty()
+        pci_devices = lspci.get_devices_by_type(
```

Comment:
The method should be like below,
1. call `_load_nics`.
2. Check the information from the class only, not call other methods. use has_device, has_module should be enough.

Reply:
updated